lib/cisco_node_utils/syslog_server.rb

```diff
@@ -0,0 +1,109 @@
+# Syslog Server provider class
+#
+# Jonathan Tripathy et al., September 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node')
+require File.join(File.dirname(__FILE__), 'interface')
+
+module Cisco
+  # NtpServer - node utility class for NTP Server configuration management
+  class SyslogServer < NodeUtil
+    attr_reader :name, :level, :vrf
+
+    def initialize(syslogserver_name,
+                   syslogserver_level=nil,
+                   syslogserver_vrf=nil,
+                   instantiate=true)
+      fail TypeError unless syslogserver_name.is_a?(String)
+      fail TypeError unless syslogserver_name.length > 0
+      @syslogserver_name = syslogserver_name
+
+      unless syslogserver_level.nil?
+        fail TypeError unless syslogserver_level.is_a?(Integer)
+      end
+      @syslogserver_level = syslogserver_level
+
+      unless syslogserver_vrf.nil?
+        fail TypeError unless syslogserver_vrf.is_a?(String)
+      end
+      @syslogserver_vrf = syslogserver_vrf
+
+      create if instantiate
+    end
+
+    def self.syslogservers
+      hash = {}
+
+      syslogservers_list = config_get('syslog_server', 'server')
+      return hash if syslogservers_list.nil?
+
+      syslogservers_list.each do |id|
+        level = config_get('syslog_server', 'level', id)
+        level = level[0].to_i unless level.nil?
+
+        vrf = config_get('syslog_server', 'vrf', id)
+        if vrf.nil?
+          vrf = 'default'
+        else
+          vrf = vrf[0]
+        end
+
+        hash[id] = SyslogServer.new(id, level, vrf, false)
+      end
+
+      hash
+    end
+
+    def ==(other)
+      name == other.name
+    end
+
+    def create
+      # Set timestamp units
+      config_set('syslog_server',
+                 'server',
+                 state: '',
+                 ip:    "#{name}",
+                 level: level.nil? ? '' : "#{level}",
+                 vrf:   vrf.nil? ? '' : "use-vrf #{vrf}",
+                )
+    end
+
+    def destroy
+      # Set timestamp units
+      config_set('syslog_server',
+                 'server',
+                 state: 'no',
+                 ip:    "#{name}",
+                 level: '',
+                 vrf:   '',
+                )
+    end
+
+    def name
```

Comment:
As noted above, if you just rename the instance variables to @name, @level, @vrf then you don't have to override the default getter methods.